metrics/metric.py

```
@@ -0,0 +1,887 @@
+from CACHE.CACHE_Config import generate_cache_file_name
+from tools.gpt_util import get_chatgpt_fields
+
+import requests
+from urllib.parse import urlencode
+import shelve
+
+S2_PAPER_URL = "https://api.semanticscholar.org/v1/paper/"
+S2_QUERY_URL = "https://api.semanticscholar.org/graph/v1/paper/search/bulk"
+CACHE_FILE = r"C:\Users\Ocean\Documents\GitHub\Dynamic_Literature_Review\CACHE\.queryCache"
+@retry()
+def request_query(query,  sort_rule=None,  continue_token=None, early_date: datetime.datetime = None, later_date:datetime.datetime = None
```

Ruff (0.8.2)

Redefinition of unused `request_query` from line 7

(F811)